.github/workflows/unit-tests.yaml

```diff
@@ -0,0 +1,50 @@
+name: Unit Tests
+
+on:
+  pull_request:
+    branches: [ main, release-* ]
+
+jobs:
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+    - name: Free some disk space on runner
+      run: |
+        echo "free space before cleanup:"
+        df -h
+        sudo rm -rf /usr/share/dotnet /opt/ghc /usr/local/share/boost /usr/lib/jvm /usr/lib/firefox /opt/microsoft/powershell /opt/hostedtoolcache
+        echo "free space after cleanup:"
+        df -h
+
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v1
+
+    - name: go cache
+      uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+
+    - name: Get dependencies
+      run: go mod download
+
+    - name: Run make test
+      run: |
+        make test COMPONENT=featuregates/controller
+        make test COMPONENT=capabilities/controller
+
+    - name: Upload coverage reports to Codecov with GitHub Action
+      uses: codecov/codecov-action@v3
+      with:
+        files: ./build/capabilities/controller/coverage/cover.out,./build/capabilities/client/coverage/cover.out,./build/featuregates/controller/coverage/cover.out,./build/featuregates/client/coverage/cover.out
```

Comment:
also, can you automate this( a script may be) to fetch all the cover.out files in the build directory? That way we don't have to worry about coverage files for new components.

Reply:
I'm afraid I don't know how to do that in a yaml file. Maybe make this incremental change and make a separate issue to try to get a script to gether all the `cover.out` files?

Reply:
thanks for the PR, I guess maybe we could also have a generic way for https://github.com/vmware-tanzu/tanzu-framework/pull/4620/files#diff-6e608e02c595d53ab6b70822a2bf19abcfc6ddcc976c2f536ad5bfca20f0443fR44-R45, so that we dont have to add another line whenever a new component get added.

Reply:
codecov action [docs](https://github.com/codecov/codecov-action) also mention a `directory` flag to search for the coverage files.

Reply:
nice find

Reply:
Yes, thank you, @avi-08! I was able to get CodeCov to search for all coverage files in the `build` directory. Check it out in my latest commit!